app/src/androidTest/java/org/akvo/flow/activity/OptionsSurveyTest.java

```diff
@@ -0,0 +1,110 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.akvo.flow.activity;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.espresso.contrib.DrawerActions;
+import android.support.test.rule.ActivityTestRule;
+
+import org.akvo.flow.R;
+import org.akvo.flow.data.database.SurveyDbAdapter;
+import org.akvo.flow.domain.Survey;
+import org.akvo.flow.testhelper.SurveyInstaller;
+import org.akvo.flow.testhelper.SurveyRequisite;
+import org.akvo.flow.util.FileUtil;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.isEnabled;
+import static android.support.test.espresso.matcher.ViewMatchers.withClassName;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.akvo.flow.tests.R.raw.optionsurvey;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.endsWith;
+
+public class OptionsSurveyTest {
+
+    @Rule
+    public ActivityTestRule<SurveyActivity> rule = new ActivityTestRule<>(SurveyActivity.class);
+    private SurveyInstaller installer;
+
+    @Before
+    public void init() {
+        Context context = rule.getActivity();
+        installer       = new SurveyInstaller(context, new SurveyDbAdapter(context));
+    }
+
+    @BeforeClass
+    public static void setRequisite() {
+        SurveyRequisite.setRequisites(InstrumentationRegistry.getTargetContext());
+    }
+
+    @After
+    public void clearSurveys() {
+        installer.clearSurveys();
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        SurveyRequisite.resetRequisites();
+    }
+
+    @Test
+    public void canFillOptionsQuestion() throws IOException, InterruptedException {
+        fillOptionsQuestion(optionsurvey, 0);
+        onView(allOf(withClassName(endsWith("Button")), withText(R.string.submitbutton))).check(matches(isEnabled()));
+    }
+
+    private Survey fillOptionsQuestion(int surveyResId, int option) throws IOException {
+        Survey survey = getSurvey(surveyResId);
+
+        openDrawer();
```

Comment:
Before selecting a survey we open the drawer. Even if it's already open, it doesn't hurt to double open it, just for the sake of assurance.